src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/util/SalesforceSplitUtil.java

```
@@ -129,6 +140,20 @@ private static BatchInfo[] runBulkQuery(BulkConnection bulkConnection, String qu
     return batchInfos;
   }
 
+  private static BatchInfo createBatchFromStream(BulkConnection bulkConnection, String query, JobInfo job) throws
+    SalesforceQueryExecutionException, IOException {
+    BatchInfo batchInfo = null;
+    try (ByteArrayInputStream bout = new ByteArrayInputStream(query.getBytes())) {
+      batchInfo = bulkConnection.createBatchFromStream(job, bout);
+    } catch (AsyncApiException exception) {
+      LOG.error("The bulk query job {} failed. Job State: {}", job.getId(), job.getState());
+      if (SalesforceBulkRecordReader.RETRY_ON_REASON.contains(exception.getExceptionCode())) {
+        throw new SalesforceQueryExecutionException(exception.getMessage());
+      }
```

if the error is not retryable, we are ignoring the error here, is it expected?

---

error logged